src/sentry/static/sentry/app/views/eventsV2/table/tableModalEditColumn.tsx

```diff
@@ -0,0 +1,237 @@
+    ),
+  };
+
+  onChangeAggregation = (value: Aggregation) => {
+    this.setState({
+      fields: filterFieldByAggregation(value),
+    });
+  };
+
+  onChangeField = (value: Field) => {
+    this.setState({
+      aggregations: filterAggregationByField(value),
+    });
+  };
+
+  onSubmitForm = (values: any) => {
+    const {indexColumnOrder, column} = this.props;
+    const {createColumn, updateColumn, onSuccess, onError} = this.props.actions;
+    const nextColumn: TableColumn<ReactText> = {...column, ...values};
+
+    try {
+      if (typeof indexColumnOrder === 'number' && this.props.column) {
+        updateColumn(indexColumnOrder, nextColumn);
+      } else {
+        createColumn(nextColumn);
+      }
+
+      if (onSuccess) {
+        onSuccess();
+      }
+    } catch (e) {
+      if (onError) {
+        onError(e);
+      }
+    }
+  };
+
+  render() {
+    const {column} = this.props;
+
+    return (
+      <React.Fragment>
+        <Form
+          onSubmit={v => this.onSubmitForm(v)}
+          submitLabel={column ? t('Update column') : t('Create column')}
+          initialData={{
+            aggregation: column ? column.aggregation : '',
+            field: column ? column.field : '',
+            name: column ? column.name : '',
+          }}
+        >
+          <FormRow>
+            <FormRowItemLeft>
+              <SelectField
+                name="aggregation"
+                label={t('Select')}
+                placeholder="Aggregate"
+                choices={this.state.aggregations}
+                onChange={this.onChangeAggregation}
+              />
+            </FormRowItemLeft>
+            <FormRowItemRight>
+              <SelectField
+                required={true}
+                name="field"
+                label={t('Column')}
+                placeholder="Column"
+                choices={this.state.fields}
+                onChange={this.onChangeField}
+              />
+            </FormRowItemRight>
+          </FormRow>
+          <FormRow>
+            <TextField
+              required={true}
+              name="name"
+              label={t('Column Name')}
+              placeholder="Column Name"
+            />
+          </FormRow>
+        </Form>
+      </React.Fragment>
+    );
+  }
+}
+
+// TODO(leedongwei): Check with Mimi to get the link for DiscoverV2 docs
+const TableModalEditColumnFooter = () => (
+  <FooterContent href="https://docs.sentry.io/">
+    <div>
+      <InlineSvg src="icon-docs" /> Discover Documentation
+    </div>
+    <div>
+      <InlineSvg src="icon-chevron-right" />
+    </div>
+  </FooterContent>
+);
+
+function filterAggregationByField(f?: Field): Aggregation[] {
+  if (!f || !FIELDS[f]) {
+    return Object.keys(AGGREGATIONS) as Aggregation[];
+  }
+
+  return Object.keys(AGGREGATIONS).reduce(
+    (accumulator, a) => {
+      if (AGGREGATIONS[a] && FIELDS[f] && AGGREGATIONS[a].type === FIELDS[f]) {
```

Comment:
Beyond the type requirements, some aggregations are only valid on certain fields. For example the `pxx` functions will only be valid on duration columns at least in the short term.